src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignmentRegion.java

```diff
@@ -0,0 +1,187 @@
+        this.assemblyId = assemblyId;
+        this.forwardStrand = alnRgn.getStrand() == '+';
+        final Cigar alignmentCigar = TextCigarCodec.decode(alnRgn.getCigar());
+        this.forwardStrandCigar = forwardStrand ? alignmentCigar : CigarUtils.invertCigar(alignmentCigar);
+        this.referenceInterval = new SimpleInterval(alnRgn.getChrom(), (int) alnRgn.getPos() + 1, (int) (alnRgn.getPos() + 1 + forwardStrandCigar.getReferenceLength()));
+        this.mqual = alnRgn.getMQual();
+        this.assembledContigLength = forwardStrandCigar.getReadLength();
+        this.startInAssembledContig = startOfAlignmentInContig(forwardStrandCigar);
+        this.endInAssembledContig = endOfAlignmentInContig(assembledContigLength, forwardStrandCigar);
+        this.mismatches = alnRgn.getNm();
+    }
+
+    public AlignmentRegion(final String assemblyId, final String contigId, final Cigar forwardStrandCigar, final boolean forwardStrand, final SimpleInterval referenceInterval, final int mqual, final int startInAssembledContig, final int endInAssembledContig, final int mismatches) {
+        this.contigId = contigId;
+        this.assemblyId = assemblyId;
+        this.forwardStrandCigar = forwardStrandCigar;
+        this.forwardStrand = forwardStrand;
+        this.referenceInterval = referenceInterval;
+        this.mqual = mqual;
+        this.startInAssembledContig = startInAssembledContig;
+        this.endInAssembledContig = endInAssembledContig;
+        this.assembledContigLength = forwardStrandCigar.getReadLength();
+        this.mismatches = mismatches;
+    }
+
+    public AlignmentRegion(final GATKRead read) {
+        this.assemblyId = null;
+        this.contigId = read.getName();
+        this.forwardStrand = ! read.isReverseStrand();
+        this.forwardStrandCigar = forwardStrand ? read.getCigar() : CigarUtils.invertCigar(read.getCigar());
+        this.referenceInterval = new SimpleInterval(read);
+        this.assembledContigLength = forwardStrandCigar.getReadLength() + getTotalHardClipping(forwardStrandCigar);
+        this.startInAssembledContig = startOfAlignmentInContig(forwardStrandCigar);
+        this.endInAssembledContig = endOfAlignmentInContig(assembledContigLength, forwardStrandCigar);
+        this.mqual = read.getMappingQuality();
+        if (read.hasAttribute("NM")) {
+            this.mismatches = read.getAttributeAsInteger("NM");
+        } else {
+            this.mismatches = 0;
+        }
+    }
+
+    public int overlapOnContig(final AlignmentRegion other) {
+        return Math.max(0, Math.min(endInAssembledContig + 1, other.endInAssembledContig + 1) - Math.max(startInAssembledContig, other.startInAssembledContig));
+    }
+
+    private static int getTotalHardClipping(final Cigar cigar) {
+        final List<CigarElement> cigarElements = cigar.getCigarElements();
+        return (cigarElements.get(0).getOperator() == CigarOperator.HARD_CLIP ? cigarElements.get(0).getLength() : 0) +
```

Comment:
doesn't handle edge cases where cigarElements.size() == 0 (probably silly)
or cigarElements.size() == 1 (which is a little less silly, but not inconceivable).